### PR TITLE
Add overradible community modal

### PR DESCRIPTION
**Pull request by jrcastro2**
* Add configurable appName for RSK
* Add configurable options to selection modal
* closes https://github.com/inveniosoftware/invenio-app-rdm/issues/2076

### CODE REVIEW

File: src/lib/components/CommunitySelectionModal/CommunitySelectionSearch.js [hunk 1]
```diff
@@ -37,13 +35,27 @@ export class CommunitySelectionSearch extends Component {
 
     this.state = {
       selectedEndpoint: defaultEndpoint,
+      overriddenComponents: props.overriddenComponents,
     };
   }
 
+  componentDidMount() {
+    const { appName } = this.props;
```
Comment (kpsherva) [inline src/lib/components/CommunitySelectionModal/CommunitySelectionSearch.js:43]: why was this moved inside the didMount? It most likely causes the components to call the render function twice

File: src/lib/components/CommunitySelectionModal/CommunitySelectionSearch.js [hunk 2]
```diff
@@ -37,13 +35,27 @@ export class CommunitySelectionSearch extends Component {
...
+    const { overriddenComponents } = this.state;
+    if (_isEmpty(overriddenComponents)) {
+      const defaultOverriddenComponents = {
+        [`${appName}.ResultsList.item`]: CommunityListItem,
+      };
+      this.setState({
+        overriddenComponents: defaultOverriddenComponents,
```
Comment (kpsherva) [inline src/lib/components/CommunitySelectionModal/CommunitySelectionSearch.js:50]: why we need overriddenComponents inside the state?

File: src/lib/components/CommunitySelectionModal/CommunitySelectionSearch.js (hunk 1 above)
Comment (jrcastro2) [reply src/lib/components/CommunitySelectionModal/CommunitySelectionSearch.js:43]: It will render twice only if we pass cmps to override the existing one, but you are right we can move this logic to the constructor itself :+1:

File: src/lib/components/CommunitySelectionModal/CommunitySelectionSearch.js (hunk 2 above)
Comment (jrcastro2) [reply src/lib/components/CommunitySelectionModal/CommunitySelectionSearch.js:50]: Because the cmp overrides cmps for RSK itself, see here: https://github.com/inveniosoftware/react-invenio-deposit/blob/master/src/lib/components/CommunitySelectionModal/CommunitySelectionSearch.js#L60

Initially we want to tune this and override this cmps to display the items differently, as in the mockups there were differences between the modal to select communities in the landing page and the modal from the deposit form, however we decided the keep them the same in the demo, so we could actually remove this.
